src/babel/transformation/transformers/es6/spec.symbols.js

```diff
@@ -8,6 +8,20 @@ export function UnaryExpression(node, parent, scope, file) {
   if (node._ignoreSpecSymbols) return;
 
   if (node.operator === "typeof") {
+    if (t.isBinaryExpression(parent)) {
+      var {operator, left, right} = parent;
+      if (right === node) {
+        [left, right] = [right, left];
+      }
+      if (t.isLiteral(right)) {
+        if ((operator === "==" || operator === "===") && right.value !== "symbol") {
+          return;
```

Comment:
Won't this give the wrong result for `typeof Symbol() === "object"` (when Symbol isn't natively supported)?